### PR TITLE
42870 v2 test linting

### PR DESCRIPTION
**Pull request by jakefhyde**
## Problem

e2e tests & v2 testing framework were not being linted

## Solution

Update `.golangci.json` config file.

## Testing

N/A

### CODE REVIEW

File: tests/v2/validation/snapshot/snapshot.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ const (
 	concurrencyDefaultValue      = "10%"
 )
 
-func SnapshotRestore(t *testing.T, client *rancher.Client, clusterName string, upgrade string, strategy bool) {
+func Restore(t *testing.T, client *rancher.Client, clusterName string, upgrade string, strategy bool) {
```
Comment (igomez06) [inline tests/v2/validation/snapshot/snapshot.go:51]: This one I don't agree with. Since this is a test func "technically" should we preserve it this way?

**Review by igomez06 — COMMENTED** [review]
looks great. I left one comment let me know what you think.

File: tests/v2/validation/snapshot/snapshot.go (hunk 1 above)
Comment (jakefhyde) [reply tests/v2/validation/snapshot/snapshot.go:51]: this comes from a linting check that exports types and functions should not have thename duplicated in the package, as for consumers it reads as `snapshot.SnapshotRestore`. What do you propose instead? I see the following options:
- `snapshot.Restore`
- `snapshot.RestoreSnapshot`
- unexporting the function to become `snapshotRestore`

Comment (igomez06) [reply tests/v2/validation/snapshot/snapshot.go:51]: This shouldn't even be exported anyway, that was a mistake. `snapshotRestore` would be best

**Review by igomez06 — APPROVED** [review]
LGTM, Jake could you do the same for release/v2.7?

**Review by slickwarren — APPROVED** [review]
lgtm

**Comment by jakefhyde** [comment]
@igomez06 Going to merge, please feel free to reach out to me directly or send others my way if any issues are encountered as a result of this. I did some basic sanity checking, but it may be a good idea to run the whole suite.

**Comment by igomez06** [comment]
> @igomez06 Going to merge, please feel free to reach out to me directly or send others my way if any issues are encountered as a result of this. I did some basic sanity checking, but it may be a good idea to run the whole suite.

sounds good!

**Pull request merged**